### PR TITLE
Adding support for GCC 7

### PR DESCRIPTION
**Pull request by elbing**
Tested: gcc-7 (Ubuntu 7-20170407-0ubuntu2)
	 7.0.1 20170407 (experimental)
	 [trunk revision 246759]

Signed-off-by: Álvaro Jurado <elbingmiss@gmail.com>

### CODE REVIEW

File: sys/src/cmd/mug.c [hunk 1]
```diff
@@ -1203,7 +1203,7 @@ main(int argc, char **argv)
 				case 10:
 					state.black += (m.xy.x-lastp.x)/255.0;
 					state.white += (m.xy.x-lastp.x)/255.0;
-					state.gamma = y2gamma(p.y);
+					//state.gamma = y2gamma(p.y); /* Has it sense? Point p never was filled and is not used anymore */
```
Comment (rminnich) [inline sys/src/cmd/mug.c]: I think this is wrong. state.gamma is now unitilized. Why not set p to be {0,0} and then use that?

Comment (elbing) [inline sys/src/cmd/mug.c]: I just was asking. Yeah why not.

**Comment by bldybuild** [comment]
Commit be3013b94188953d208706bcf91488c38c9e9f4d is not signed-off.

Please fix these issues:

>	1. Álvaro Jurado has not signed-off be3013b.


If you'd like more information on how to sign your commits please visit [signed-off-by.me](https://signed-off-by.me)

**Comment by elbing** [comment]
This leaves PR only with gcc-7.json changes. Until we have a solution for macOS PIE issue.

**Comment by elbing** [comment]
I'll rebase it then. Thanks Ron.

**Pull request merged**